File: 1598.py
text=input()
a, b = text.split(" ")
a = int(a)
b = int(b)
a_x = a % 4
a_y = (a//4)
if(a_x == 0):
	a_x = 4
else:
	a_y += 1
b_x = b % 4
b_y = (b // 4)
if(b_x == 0):
	b_x = 4
else:
	b_y += 1
print (abs(b_x - a_x)+abs(b_y - a_y))

File: 1598.cpp
#include <stdio.h>

int abs(int a){
	if (a > 0)
		return a;
	else
		return -a;
}

int p(int a){
	if(a % 4 == 0)
		return 4;
	else
		return a % 4;
}

int n(int a){
	if(p(a) == 4)
		return a/4;
	else
		return (a/4)+1;
}

int main(void){
	int a[2];
	scanf("%d %d", &a[0], &a[1]);
	printf("%d\n", abs(p(a[0])-p(a[1])) + abs(n(a[0]) - n(a[1])));
	return 0;
}
